ext/pdo/pdo_stmt.c

```diff
@@ -750,7 +750,7 @@ static int do_fetch(pdo_stmt_t *stmt, int do_bind, zval *return_value, enum pdo_
 {
 	int flags, idx, old_arg_count = 0;
 	zend_class_entry *ce = NULL, *old_ce = NULL;
-	zval grp_val, *pgrp, retval, old_ctor_args = {{0}, {{0}}, {0}};
```

Comment:
This was recently added in https://github.com/php/php-src/commit/6e40ec7de95a945a255f7d16b8ead51ad68e9bf5 to fix another warning... cc @Girgias

Reply:
This may be a bug in GCC as to why it reports the missing initializer after digging around a bit:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=80454
I wonder if just doing ``{0}`` would fix both warnings as this should NULL initialize the whole struct from my (albeit limited) understanding.
Will try to apply this change and run on CI in #5151

Reply:
Wouldn't it be cleaner to omit the initializer, and to do `ZVAL_UNDEF(old_ctor_args)` instead?

Reply:
@cmb69 You won't be believe me, but doing that is going to trigger *another* compiler warning (`-Wmaybe-uninitialized`). No matter what you do, there's always a warning...

Reply:
So I tried just using `{0}` and Clang complains...

Reply:
sigh

Reply:
So got a fix which doesn't emit a warning on *any* platform: https://github.com/php/php-src/pull/5151/commits/31f3b78ece0905584163ca724cdc9f6e875abe81
Many thanks to @booti386 for helping me out here.

Reply:
@Girgias The above mentioned fix throws error on little as well as big endian arch without removing the `--enable-werror` flag from travis/compile.sh.